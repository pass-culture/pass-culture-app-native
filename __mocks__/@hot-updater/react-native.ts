import { ReactNode } from 'react'

const getUpdateSource = jest.fn(() => Promise.resolve({ source: 'source' }))

const HotUpdater = jest.fn(() => (app: ReactNode) => app)

export default Object.assign(HotUpdater, {
  getUpdateSource,
})
