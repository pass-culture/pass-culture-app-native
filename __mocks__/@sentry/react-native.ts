import * as ActualSentryModule from '@sentry/react-native'

const MockedSentryModule = jest.genMockFromModule<typeof ActualSentryModule>('@sentry/react-native')

module.exports = {
  ...MockedSentryModule,
}
