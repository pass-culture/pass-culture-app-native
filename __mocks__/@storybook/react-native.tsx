import React from 'react'; // @storybook
import { View } from 'react-native';

const configure = jest.fn();
const getStorybookUI = jest.fn().mockImplementation(() => () => <View />);

export { configure, getStorybookUI };
