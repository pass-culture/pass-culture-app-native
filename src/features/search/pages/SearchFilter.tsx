import { t } from '@lingui/macro'
import React from 'react'
import { Dimensions } from 'react-native'
import styled from 'styled-components/native'

import { useUserProfileInfo } from 'features/home/api'
import { ShowResults, ReinitializeFilters } from 'features/search/atoms/Buttons'
import { useSearch } from 'features/search/pages/SearchWrapper'
import Section from 'features/search/sections'
import { _ } from 'libs/i18n'
import { PageHeader } from 'ui/components/headers/PageHeader'
import { ColorsEnum, getSpacing, Spacer } from 'ui/theme'

export const SearchFilter: React.FC = () => {
  const { searchState } = useSearch()
  const { data: profile } = useUserProfileInfo()

  return (
    <React.Fragment>
      <React.Fragment>
        <Container>
          <Spacer.TopScreen />
          <Spacer.Column numberOfSpaces={16} />

          {/* Localisation */}
          <Section.Location />
          <Separator marginVertical={getSpacing(6)} />

          {/* Rayon */}
          <Section.Radius />
          <Spacer.Column numberOfSpaces={6} />
          <Separator />

          {/* Catégories */}
          <Section.Category />
          <Spacer.Column numberOfSpaces={2} />
          <Separator />

          {/* Type d'offre */}
          <Section.OfferType />
          <Spacer.Column numberOfSpaces={2} />
          <Separator />
          <Spacer.Column numberOfSpaces={6} />

          {/* Prix */}
          <Section.Price />
          <Separator marginVertical={getSpacing(6)} />

          {/* Uniquement les offres gratuites */}
          <Section.FreeOffer />
          <Separator marginVertical={getSpacing(6)} />

          {/* Uniquement les offres duo */}
          {profile?.isBeneficiary && (
            <React.Fragment>
              <Section.DuoOffer />
              <Separator marginVertical={getSpacing(6)} />
            </React.Fragment>
          )}

          {/* Uniquement les nouveautés */}
          <Section.NewOffer />
          <Separator marginVertical={getSpacing(6)} />

          {/* Date */}
          <Section.Date />
          <Separator marginVertical={getSpacing(6)} />

          {/* Date de l'offre */}
          {!!searchState.date && (
            <React.Fragment>
              <Section.OfferDate />
              <Separator marginVertical={getSpacing(6)} />
            </React.Fragment>
          )}

          {/* Heure */}
          <Section.Hour />

          {/*Créneau horaire */}
          {!!searchState.timeRange && (
            <React.Fragment>
              <Separator marginVertical={getSpacing(6)} />
              <Section.TimeSlot />
            </React.Fragment>
          )}

          <Spacer.Column numberOfSpaces={30} />
        </Container>
      </React.Fragment>

      <PageHeader title={_(t`Filtrer`)} RightComponent={ReinitializeFilters} />

      <ShowResultsContainer>
        <ShowResults />
        <Spacer.BottomScreen />
      </ShowResultsContainer>
    </React.Fragment>
  )
}

const { width } = Dimensions.get('window')
const Container = styled.ScrollView({ flex: 1 })
const Separator = styled.View<{ marginVertical?: number }>(({ marginVertical = 0 }) => ({
  width: width - getSpacing(2 * 6),
  height: 2,
  backgroundColor: ColorsEnum.GREY_LIGHT,
  alignSelf: 'center',
  marginVertical: marginVertical,
}))
const ShowResultsContainer = styled.View({
  width: '100%',
  position: 'absolute',
  bottom: getSpacing(6),
  paddingHorizontal: getSpacing(6),
})
