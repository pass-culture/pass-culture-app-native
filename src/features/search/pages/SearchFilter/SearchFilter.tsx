import { useNavigation, useRoute } from '@react-navigation/native'
import React, { useCallback, useEffect, useMemo } from 'react'
import { ScrollView } from 'react-native'
import { useTheme } from 'styled-components'
import styled from 'styled-components/native'

import { useAuthContext } from 'features/auth/context/AuthContext'
import { UseNavigationType, UseRouteType } from 'features/navigation/RootNavigator/types'
import { getTabNavConfig } from 'features/navigation/TabBar/helpers'
import { FilterPageButtons } from 'features/search/components/FilterPageButtons/FilterPageButtons'
import Section from 'features/search/components/sections'
import { DEFAULT_RADIUS } from 'features/search/constants'
import { useSearch } from 'features/search/context/SearchWrapper'
import { FilterBehaviour, LocationType } from 'features/search/enums'
import { LocationFilter, SearchView } from 'features/search/types'
import { analytics } from 'libs/analytics'
import { useLocation } from 'libs/geolocation'
import { useFunctionOnce } from 'libs/hooks'
import { BlurHeader } from 'ui/components/headers/BlurHeader'
import {
  PageHeaderWithoutPlaceholder,
  useGetHeaderHeight,
} from 'ui/components/headers/PageHeaderWithoutPlaceholder'
import { Li } from 'ui/components/Li'
import { VerticalUl } from 'ui/components/Ul'
import { getSpacing, Spacer } from 'ui/theme'

export const SearchFilter: React.FC = () => {
  const headerHeight = useGetHeaderHeight()
  const { searchState, dispatch } = useSearch()
  const { navigate } = useNavigation<UseNavigationType>()
  const logReinitializeFilters = useFunctionOnce(() => {
    analytics.logReinitializeFilters(searchState.searchId)
  })
  const { userPosition: position, isGeolocated, isCustomPosition, place } = useLocation()
  const { user } = useAuthContext()
  const { params } = useRoute<UseRouteType<'SearchFilter'>>()
  const { isMobileViewport } = useTheme()

  useEffect(() => {
    dispatch({ type: 'SET_STATE', payload: params || { view: SearchView.Landing } })
  }, [dispatch, params])

  const onGoBack = useCallback(() => {
    navigate(
      ...getTabNavConfig('Search', {
        ...params,
        view: SearchView.Results,
      })
    )
  }, [navigate, params])

  const onSearchPress = useCallback(() => {
    const additionalSearchState = {
      ...searchState,
      view: SearchView.Results,
    }
    navigate(...getTabNavConfig('Search', additionalSearchState))
  }, [navigate, searchState])

  const onResetPress = useCallback(() => {
    const getLocationFilter = (): LocationFilter => {
      const { locationFilter } = searchState
      const aroundRadius =
        locationFilter.locationType === LocationType.EVERYWHERE ||
        locationFilter.locationType === LocationType.VENUE
          ? DEFAULT_RADIUS
          : locationFilter.aroundRadius
      if (isCustomPosition && place) {
        return {
          locationType: LocationType.PLACE,
          place,
          aroundRadius: aroundRadius ?? DEFAULT_RADIUS,
        }
      } else if (isGeolocated && position) {
        return {
          locationType: LocationType.AROUND_ME,
          aroundRadius,
        }
      } else {
        return { locationType: LocationType.EVERYWHERE }
      }
    }
    dispatch({
      type: 'SET_STATE',
      payload: {
        ...searchState,
        locationFilter: getLocationFilter(),
        minPrice: undefined,
        maxPrice: undefined,
        offerGenreTypes: undefined,
        offerNativeCategories: undefined,
        beginningDatetime: undefined,
        date: null,
        endingDatetime: undefined,
        offerCategories: [],
        offerSubcategories: [],
        offerIsDuo: false,
        offerIsFree: false,
        offerIsNew: false,
        priceRange: null,
        tags: [],
        timeRange: null,
      },
    })
    logReinitializeFilters()
  }, [
    dispatch,
    isCustomPosition,
    isGeolocated,
    logReinitializeFilters,
    place,
    position,
    searchState,
  ])

  const hasDuoOfferToggle = useMemo(() => {
    const isBeneficiary = !!user?.isBeneficiary
    const hasRemainingCredit = !!user?.domainsCredit?.all?.remaining

    return isBeneficiary && hasRemainingCredit
  }, [user?.isBeneficiary, user?.domainsCredit?.all?.remaining])

  const shouldDisplayCloseButton = isMobileViewport

  const sectionItems = [Section.Category]
  if (hasDuoOfferToggle) sectionItems.push(Section.OfferDuo)
  sectionItems.push(Section.Venue)
  sectionItems.push(Section.Price)
  sectionItems.push(Section.DateHour)

  return (
    <Container>
      <PageHeaderWithoutPlaceholder title="Filtres" onGoBack={onGoBack} />

      <ScrollView scrollEnabled keyboardShouldPersistTaps="always">
        <Placeholder height={headerHeight} />
        <VerticalUl>
          <Spacer.Column numberOfSpaces={2} />
          {sectionItems.map((SectionItem, index) => {
            return (
              <SectionWrapper key={index} isFirstSectionItem={index === 0}>
                <SectionItem onClose={shouldDisplayCloseButton ? onGoBack : undefined} />
              </SectionWrapper>
            )
          })}
        </VerticalUl>
      </ScrollView>
      <BlurHeader height={headerHeight} />
      <Spacer.Column numberOfSpaces={10} />
      <FilterPageButtons
        onResetPress={onResetPress}
        onSearchPress={onSearchPress}
        isModal={false}
        filterBehaviour={FilterBehaviour.SEARCH}
      />
    </Container>
  )
}

const SectionWrapper: React.FunctionComponent<{
  children: React.JSX.Element
  isFirstSectionItem: boolean
}> = ({ children, isFirstSectionItem = false }) => {
  return (
    <StyledLi>
      {isFirstSectionItem ? null : <Separator />}
      <Spacer.Column numberOfSpaces={6} />
      {children}
      <Spacer.Column numberOfSpaces={6} />
    </StyledLi>
  )
}

const Container = styled.View(({ theme }) => ({
  backgroundColor: theme.colors.white,
}))

const Separator = styled.View(({ theme }) => ({
  width: '100%',
  height: 2,
  backgroundColor: theme.colors.greyLight,
}))

const StyledLi = styled(Li)({
  display: 'flex',
  marginLeft: getSpacing(6),
  marginRight: getSpacing(6),
})

const Placeholder = styled.View<{ height: number }>(({ height }) => ({
  height,
}))
