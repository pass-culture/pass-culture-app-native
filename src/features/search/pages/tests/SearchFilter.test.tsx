import { render } from '@testing-library/react-native'
import React from 'react'

import { SearchFilter } from '../SearchFilter'

describe('SearchFilter component', () => {
  it('should render correctly', () => {
    const { toJSON } = render(<SearchFilter />)
    expect(toJSON()).toMatchSnapshot()
  })
})
