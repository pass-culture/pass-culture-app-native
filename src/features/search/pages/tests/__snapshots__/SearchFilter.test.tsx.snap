// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchFilter component should render correctly 1`] = `
<RCTScrollView
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "marginHorizontal": 24,
      },
    ]
  }
>
  <View>
    <View
      customHeight={16}
      style={
        Array [
          Object {
            "height": 16,
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#151515",
            "fontFamily": "Montserrat-Medium",
            "fontSize": 18,
            "lineHeight": 22,
          },
        ]
      }
    >
      Rayon
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "marginTop": 16,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#151515",
              "fontFamily": "Montserrat-Bold",
              "fontSize": 15,
              "lineHeight": 20,
            },
          ]
        }
      >
        100 km
      </Text>
      <View
        numberOfSpaces={4}
        style={
          Array [
            Object {
              "height": 16,
            },
          ]
        }
      />
      <View>
        <View
          style={
            Array [
              Object {
                "height": 50,
                "justifyContent": "center",
                "position": "relative",
              },
              Object {
                "height": 20,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                Object {
                  "width": 280,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#A7A7A7",
                    "borderRadius": 2,
                    "height": 2,
                  },
                  Object {
                    "borderRadius": 24,
                    "height": 15,
                    "marginTop": -7,
                  },
                  Object {
                    "backgroundColor": "#eb0055",
                  },
                  Object {
                    "width": 280,
                  },
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#A7A7A7",
                    "borderRadius": 2,
                    "height": 2,
                  },
                  Object {
                    "borderRadius": 24,
                    "height": 15,
                    "marginTop": -7,
                  },
                  Object {
                    "backgroundColor": "#c7c7cc",
                  },
                  Object {
                    "width": 0,
                  },
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "height": 48,
                    "justifyContent": "center",
                    "position": "absolute",
                    "width": 48,
                  },
                  Object {
                    "left": 256,
                    "top": -24,
                  },
                  undefined,
                  Object {
                    "zIndex": 1,
                  },
                ]
              }
            >
              <View
                onMoveShouldSetResponder={[Function]}
                onMoveShouldSetResponderCapture={[Function]}
                onResponderEnd={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderReject={[Function]}
                onResponderRelease={[Function]}
                onResponderStart={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                onStartShouldSetResponderCapture={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "alignSelf": "stretch",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    },
                    Object {
                      "borderRadius": 15,
                    },
                  ]
                }
              >
                <View
                  accessible={true}
                  focusable={false}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                >
                  <View
                    style={
                      Array [
                        Object {
                          "backgroundColor": "#FFFFFF",
                          "borderColor": "#DDDDDD",
                          "borderRadius": 30,
                          "borderWidth": 1,
                          "height": 30,
                          "shadowColor": "#000000",
                          "shadowOffset": Object {
                            "height": 3,
                            "width": 0,
                          },
                          "shadowOpacity": 0.2,
                          "shadowRadius": 1,
                          "width": 30,
                        },
                        Object {
                          "backgroundColor": "#ffffff",
                          "borderColor": "#ffffff",
                          "borderRadius": 28,
                          "elevation": 4,
                          "height": 28,
                          "shadowColor": "#151515",
                          "shadowOpacity": 0.2,
                          "shadowRadius": 4,
                          "width": 28,
                        },
                        undefined,
                        undefined,
                      ]
                    }
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "#f5f5f5",
            "height": 2,
            "marginVertical": 24,
            "width": "100%",
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#151515",
            "fontFamily": "Montserrat-Medium",
            "fontSize": 18,
            "lineHeight": 22,
          },
        ]
      }
    >
      Prix
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "marginTop": 16,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#151515",
              "fontFamily": "Montserrat-Bold",
              "fontSize": 15,
              "lineHeight": 20,
            },
          ]
        }
      >
        0 € - 300 €
      </Text>
      <View
        numberOfSpaces={4}
        style={
          Array [
            Object {
              "height": 16,
            },
          ]
        }
      />
      <View>
        <View
          style={
            Array [
              Object {
                "height": 50,
                "justifyContent": "center",
                "position": "relative",
              },
              Object {
                "height": 20,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                Object {
                  "width": 280,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#A7A7A7",
                    "borderRadius": 2,
                    "height": 2,
                  },
                  Object {
                    "borderRadius": 24,
                    "height": 15,
                    "marginTop": -7,
                  },
                  Object {
                    "backgroundColor": "#c7c7cc",
                  },
                  Object {
                    "width": 0,
                  },
                ]
              }
            />
            <View
              onMoveShouldSetResponder={[Function]}
              onMoveShouldSetResponderCapture={[Function]}
              onResponderEnd={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderReject={[Function]}
              onResponderRelease={[Function]}
              onResponderStart={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              onStartShouldSetResponderCapture={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "#A7A7A7",
                    "borderRadius": 2,
                    "height": 2,
                  },
                  Object {
                    "borderRadius": 24,
                    "height": 15,
                    "marginTop": -7,
                  },
                  Object {
                    "backgroundColor": "#eb0055",
                  },
                  Object {
                    "width": 280,
                  },
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#A7A7A7",
                    "borderRadius": 2,
                    "height": 2,
                  },
                  Object {
                    "borderRadius": 24,
                    "height": 15,
                    "marginTop": -7,
                  },
                  Object {
                    "backgroundColor": "#c7c7cc",
                  },
                  Object {
                    "width": 0,
                  },
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "height": 48,
                    "justifyContent": "center",
                    "position": "absolute",
                    "width": 48,
                  },
                  Object {
                    "left": -24,
                    "top": -24,
                  },
                  undefined,
                  false,
                ]
              }
            >
              <View
                onMoveShouldSetResponder={[Function]}
                onMoveShouldSetResponderCapture={[Function]}
                onResponderEnd={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderReject={[Function]}
                onResponderRelease={[Function]}
                onResponderStart={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                onStartShouldSetResponderCapture={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "alignSelf": "stretch",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    },
                    Object {
                      "borderRadius": 15,
                    },
                  ]
                }
              >
                <View
                  accessible={true}
                  focusable={false}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                >
                  <View
                    style={
                      Array [
                        Object {
                          "backgroundColor": "#FFFFFF",
                          "borderColor": "#DDDDDD",
                          "borderRadius": 30,
                          "borderWidth": 1,
                          "height": 30,
                          "shadowColor": "#000000",
                          "shadowOffset": Object {
                            "height": 3,
                            "width": 0,
                          },
                          "shadowOpacity": 0.2,
                          "shadowRadius": 1,
                          "width": 30,
                        },
                        Object {
                          "backgroundColor": "#ffffff",
                          "borderColor": "#ffffff",
                          "borderRadius": 28,
                          "elevation": 4,
                          "height": 28,
                          "shadowColor": "#151515",
                          "shadowOpacity": 0.2,
                          "shadowRadius": 4,
                          "width": 28,
                        },
                        undefined,
                        undefined,
                      ]
                    }
                  />
                </View>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "height": 48,
                    "justifyContent": "center",
                    "position": "absolute",
                    "width": 48,
                  },
                  Object {
                    "right": -24,
                    "top": -24,
                  },
                  undefined,
                ]
              }
            >
              <View
                onMoveShouldSetResponder={[Function]}
                onMoveShouldSetResponderCapture={[Function]}
                onResponderEnd={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderReject={[Function]}
                onResponderRelease={[Function]}
                onResponderStart={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                onStartShouldSetResponderCapture={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "alignSelf": "stretch",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    },
                    Object {
                      "borderRadius": 15,
                    },
                  ]
                }
              >
                <View
                  accessible={true}
                  focusable={false}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                >
                  <View
                    style={
                      Array [
                        Object {
                          "backgroundColor": "#FFFFFF",
                          "borderColor": "#DDDDDD",
                          "borderRadius": 30,
                          "borderWidth": 1,
                          "height": 30,
                          "shadowColor": "#000000",
                          "shadowOffset": Object {
                            "height": 3,
                            "width": 0,
                          },
                          "shadowOpacity": 0.2,
                          "shadowRadius": 1,
                          "width": 30,
                        },
                        Object {
                          "backgroundColor": "#ffffff",
                          "borderColor": "#ffffff",
                          "borderRadius": 28,
                          "elevation": 4,
                          "height": 28,
                          "shadowColor": "#151515",
                          "shadowOpacity": 0.2,
                          "shadowRadius": 4,
                          "width": 28,
                        },
                        undefined,
                        undefined,
                      ]
                    }
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
